In [1]:
import numpy as np

# Problema 1

Las matrices $A$ y $B$ representan las ventas en dos años, en millones de dólares, de los tres productos de una empresa en las cuatro regiones en que está disponible. Es decir, $a_{ij}$ y $b_{ij}$ representan el monto de las ventas del producto $i$ en la región $j$ el primer y segundo año, respectivamente:

<br>

\begin{split}
A = \begin{pmatrix}
 2,6 &  4,8 &  1,8 & 0,9 \\
 3,2 &  4,4 &  2,5 & 2,8 \\
 2,4 &  3,6 &  3,8 & 2,5 \\
\end{pmatrix} \text{ } B= \begin{pmatrix}
 3,6 &  2,5 &  3,0 & 2,5 \\
 4,5 &  5,0 &  3,5 & 3,8 \\
 2,9 &  3,0 &  4,6 & 4,0 \\
\end{pmatrix}
\end{split}

<br>

1. Indique el orden de cada matriz.
2. Calcule la matriz $S = A + B$.
3. Calcule la matriz $D = B - A$.
4. Indique el valor e interprete los elementos $s_{24}$ y $d_{24}$.
5. Para el tercer año de funcionamiento, los tres productos ahora están disponibles en seis regiones. Durante este año, las ventas de esta empresa estarán registradas en una matriz $E$. ¿Es posible determinar la matriz $T = A + B + E$? Justifique su respuesta.

# Solución

In [5]:
A = np.array([[2.6, 4.8, 1.8, 0.9],
              [3.2, 4.4, 2.5, 2.8],
              [2.4, 3.6, 3.8, 2.5]])

B = np.array([[3.6, 2.5, 3.0, 2.5],
              [4.5, 5.0, 3.5, 3.8],
              [2.9, 3.0, 4.6, 4.0]])

**1)**

Ambas matrices son de orden $3$ x $4$

**2)**

In [8]:
S = A + B
for line in S:
    print("  ".join([str(round(n,2 )) for n in line]))

6.2  7.3  4.8  3.4
7.7  9.4  6.0  6.6
5.3  6.6  8.4  6.5


**3)**

In [9]:
D = B - A
for line in D:
    print("  ".join(str(round(n, 2)) for n in line))

1.0  -2.3  1.2  1.6
1.3  0.6  1.0  1.0
0.5  -0.6  0.8  1.5


**4)**

In [11]:
print(f"S_24 = {S[1][3]}")
print(f"D_24 = {D[1][3]}")

S_24 = 6.6
D_24 = 1.0


Se han vendido en total $6.6$ millones de dólares del producto 2 en la región 4 en los dos años.

Han aumentado en $1$ millón de dolares la venta del producto 2 en la región 4 en el segundo año con respecto al primer año.


**5)**

# Problema 2

Una fábrica con sucursales en Japón y China, produce tres tipos de componentes $j$, para cinco modelos de computador $i$. Los costos de producción mensual, en miles de dólares, de ambas sucursales se representan en las matrices $J$ y $C$, respectivamente:

<br>

\begin{split}
J = \begin{pmatrix}
 100 &  200 &  150  \\
 120 &  210 &  170  \\
 112 &  200 &  131  \\
 124 &  310 &  170  \\
 136 &  156 &  140  \\
\end{pmatrix} \text{ } C = \begin{pmatrix}
 110 &  190 &  160  \\
 130 &  220 &  180  \\
 146 &  190 &  141  \\
 143 &  312 &  163  \\
 137 &  116 &  156  \\
\end{pmatrix}
\end{split}

<br>

1. ¿Cuál es el orden de las matrices?
2. Inteprete el elemento $j_{21}$ y $c_{12}$ en el contexto del problema.
3. Utilizando *Python*, calcule el costo total de producción para cada modelo, considerando ambas sucursales.
4. Un trabajador indica que, en la sucursal japonesa, el costo de producción del producto $3$, modelo $4$ de computador, es más alto que en la sucursal china. ¿Tiene razón? Justifique apropiadamente.

# Solución

In [12]:
J = np.array([[100, 200, 150],
              [120, 210, 170],
              [112, 200, 131],
              [124, 310, 170],
              [136, 156, 140]])

C = np.array([[110, 190, 160],
              [130, 220, 180],
              [146, 190, 141],
              [143, 312, 163],
              [137, 116, 156]])

**1)**

El orden de las matrices es $5$ x $3$

**2)**

In [13]:
print(f"J_21 = {J[1][0]}")
print(f"J_12 = {C[0][1]}")

J_21 = 120
J_12 = 190


Mensualmente cuesta $120$ mil dólares producir el primer componente del segundo modelo de computador en Japón.

Mensualmente cuesta $190$ mil dólares producir el segundo componente del primer modelo de computador en China.

**3)**

In [22]:
S = J + C 

print("Suma de J + C: ")
for line in S:
    print("  ".join(str(n) for n in line))

print()
for i in range (0,5):
    print(f"El costo total del modelo {i+1} es {sum(S[i])} miles de dólares")

Suma de J + C: 
210  390  310
250  430  350
258  390  272
267  622  333
273  272  296

El costo total del modelo 1 es 910 miles de dólares
El costo total del modelo 2 es 1030 miles de dólares
El costo total del modelo 3 es 920 miles de dólares
El costo total del modelo 4 es 1222 miles de dólares
El costo total del modelo 5 es 841 miles de dólares


**4)**

In [23]:
print(f"J_43 = {J[3][2]}")
print(f"C_43 = {C[3][2]}")

J_43 = 170
C_43 = 163


El trabajador tiene la razón

# Problema 3

La empresa Beta tiene cinco plantas en Chile, y en cada una de ellas fabrica siete productos. El número de unidades del producto $i$ fabricadas en la planta $j$ en una semana está representado en la matriz $C$ en el elemento $c_{ij}$.

<br>

\begin{split}
C = \begin{pmatrix}
 120 &  200 &  155 &  125  &  230 \\
 155 &  130 &  110 &  160  &  180 \\
 140 &  205 &  150 &  125  &  155 \\
 130 &  120 &  230 &  225  &  125 \\
 135 &  70  &  190 &  175  &  95  \\
 110 &  265 &  180 &  230  &  180 \\
 185 &  85  &  120 &  135  &  150 \\
\end{pmatrix}
\end{split}

<br>

La empresa se expande, creando cinco plantas en Ecuador que producen los mismos siete productos. La producción semanal de esta filial estará representada por una matriz $E$.

1. Si se sabe que los volúmenes de producción semanal de la filial en Ecuador son un $20\%$ menor que en Chile, calcule la matriz $E$ utilizando *Python*.
2. Si representamos la producción semanal total de la empresa Beta, considerando ambas filiales, por la matriz $T$, escriba la expresión que calcula cada elemento de la matriz $T$ usando notación matricial.
3. Calcule la matriz $T$ utilizando *Python*.
4. Interprete los elementos $c_{34}$ y $t_{72}$ en el contexto del problema.

# Solución

In [24]:
C = np.array([[120, 200, 155, 125, 230],
              [155, 130, 110, 160, 180],
              [140, 205, 150, 125, 155],
              [130, 120, 230, 225, 125],
              [135, 70, 190, 175, 95],
              [110, 265, 180, 230, 180],
              [185, 85, 120, 135, 150]])

**1)**

In [28]:
E = 0.8 * C

for line in E:
    print(" ".join(str(round(n)).rjust(3, " ") for n in line))

 96 160 124 100 184
124 104  88 128 144
112 164 120 100 124
104  96 184 180 100
108  56 152 140  76
 88 212 144 184 144
148  68  96 108 120


**2)**

$T_{ij} = C_{ij} + E_{ij}$

**3)**

In [30]:
T = C + E

for line in T:
    print("  ".join(str(round(n)) for n in line))

216  360  279  225  414
279  234  198  288  324
252  369  270  225  279
234  216  414  405  225
243  126  342  315  171
198  477  324  414  324
333  153  216  243  270


**4)**

In [31]:
print(f"C_34 = {C[2][3]}")
print(f"T_72 = {T[6][1]}")

C_34 = 125
T_72 = 153.0


En Chile se producen $125$ unidades del producto 3 en la planta 4
En total la empresa produce $153$ unidades del producto 7 en la planta 2

# Problema 4

Una empresa que fabrica zapatos tiene dos plantas, una en Valparaíso y la otra en Temuco. La empresa produce zapatos de color negro, blanco y café, tanto para niños, damas como caballeros. La capacidad actual de producción mensual (en miles de pares) para Valparaíso y Temuco, están dadas por las matrices $V$ y $T$, respectivamente. Para ambas matrices, según el orden en que se mencionan, las filas corresponden al color del calzado y las columnas al tipo de persona que utilizará el calzado.

<br>

\begin{split}
V = \begin{pmatrix}
 30 &  34 &  20  \\
 45 &  20 &  16  \\
 14 &  26 &  25  \\
\end{pmatrix} \text{ } T = \begin{pmatrix}
 35 &  30 &  26  \\
 52 &  25 &  18  \\
 23 &  24 &  32  \\
\end{pmatrix}
\end{split}

<br>

1. Determine la producción mensual total de ambas plantas, representándola en una matriz $A$. Utilice *Python* para resolver.
2. Interprete el elemento $a_{21}$ en el contexto del problema.
3. Si para el próximo año se estima que la producción en Valparaíso aumenta en un $30\%$, mientras que la de Temuco se reduce en un $10\%$, represente la nueva producción total en la matriz $B$.
4. Interprete el elemento $b_{13}$ en el contexto del problema.

# Solución

In [32]:
V = np.array([[30, 34, 20],
              [45, 20, 16],
              [14, 26, 25]])

T = np.array([[35, 30, 26],
              [52, 25, 18],
              [23, 24, 32]])

**1)**

In [36]:
A = V + T

print("Matriz resultante A:\n")
for line in A:
    print(" ".join(str(n) for n in line))

Matriz resultante A:

65 64 46
97 45 34
37 50 57


**2)**

In [37]:
print(f"A_21 = {A[1][0]}")

A_21 = 97


La empresa fabrica mensualmente $97$ miles de pares de zapatos blancos para niños entre todas sus plantas.

**3)**

In [42]:
V2 = V * 1.3
T2 = T * 0.9

B = V2 + T2

for line in B:
    print("  ".join(str(round(n, 2)).rjust(5, " ") for n in line))

 70.5   71.2   49.4
105.3   48.5   37.0
 38.9   55.4   61.3


**4)**

In [45]:
print(f"B_13 = {round(B[0][2],2)}")

B_13 = 49.4


El próximo año se estima que en total la empresa frabricará $49.4$ miles de pares de zapatos negros para caballeros.

# Problema 5

Genere una matriz $M$ de orden $8 \times 9$ donde cada término de la matriz está definido por:

$$m_{ij} = 8i + 12j$$

y calcule su transpuesta.

# Solución

In [62]:
M = np.fromfunction(lambda i,j: 8*(i+1)+ 12* (j+1), (8,9))

print("Matriz M: \n")
for line in M:
    print("  ".join(str(round(n)).rjust(3, " ") for n in line))

MT = M.T

print("\n\nMatriz M transpuesta: \n")
for line in MT:
    print("  ".join(str(round(n)).rjust(3, " ") for n in line))

Matriz M: 

 20   32   44   56   68   80   92  104  116
 28   40   52   64   76   88  100  112  124
 36   48   60   72   84   96  108  120  132
 44   56   68   80   92  104  116  128  140
 52   64   76   88  100  112  124  136  148
 60   72   84   96  108  120  132  144  156
 68   80   92  104  116  128  140  152  164
 76   88  100  112  124  136  148  160  172


Matriz M transpuesta: 

 20   28   36   44   52   60   68   76
 32   40   48   56   64   72   80   88
 44   52   60   68   76   84   92  100
 56   64   72   80   88   96  104  112
 68   76   84   92  100  108  116  124
 80   88   96  104  112  120  128  136
 92  100  108  116  124  132  140  148
104  112  120  128  136  144  152  160
116  124  132  140  148  156  164  172


# Problema 6

Genere dos matrices:

*   Una matriz $A$ de orden $5 \times 7$, con $a_{ij} = i \cdot j.$
*   Una matriz $B$ de orden $7 \times 5$, con $b_{ij} = 15i - 2j.$

y calcule $C = A^{T} + \dfrac{1}{2}B$.



# Solución

In [67]:
A = np.fromfunction(lambda i,j: (i+1)*(j+1), (5,7))
B = np.fromfunction(lambda i,j: 15*(i+1)- 2*(j+1), (7,5))

print("Matriz A:\n")
for line in A:
    print("  ".join(str(round(n)).rjust(2, " ") for n in line))

print("\n\n")

print("Matriz B:\n")
for line in B:
    print("  ".join(str(round(n)).rjust(3, " ") for n in line))

Matriz A:

 1   2   3   4   5   6   7
 2   4   6   8  10  12  14
 3   6   9  12  15  18  21
 4   8  12  16  20  24  28
 5  10  15  20  25  30  35



Matriz B:

 13   11    9    7    5
 28   26   24   22   20
 43   41   39   37   35
 58   56   54   52   50
 73   71   69   67   65
 88   86   84   82   80
103  101   99   97   95


In [70]:
C = A.T + (0.5) * B

print("Matriz C:\n")
for line in C:
    print("  ".join(str(round(n,2)).rjust(5, " ") for n in line))

Matriz C:

  7.5    7.5    7.5    7.5    7.5
 16.0   17.0   18.0   19.0   20.0
 24.5   26.5   28.5   30.5   32.5
 33.0   36.0   39.0   42.0   45.0
 41.5   45.5   49.5   53.5   57.5
 50.0   55.0   60.0   65.0   70.0
 58.5   64.5   70.5   76.5   82.5


# Problema 7

Una red de sensores recoge datos de temperatura ($°C$), humedad (porcentaje) y presión (hPa) en cinco días distintos (de lunes a viernes). Los datos medidos se muestran en la matriz $D$:

<br>

\begin{equation*}
\begin{matrix}
& \begin{matrix}
 & Lu  \ \ &Ma \ \ &Mi \ \ & Ju \ \ &Vi
\end{matrix} \\
&D = \begin{pmatrix}
20    &  28    &  22    &  26     &  24   \\
52    &  57    &  56    &  54     &  58   \\
1.066 &  1.068 &  1.066 &  1.064  &  1.068 \\
\end{pmatrix} & \ \begin{matrix}
Cº\\
\% \\
hPa
\end{matrix}\\
\end{matrix}
\end{equation*}

<br>

Se sabe que por un error de calibración de los sensores, las medidas registradas deberían ser un $5\%$ menores que las registradas.

1. Calcule en una matriz $S$ los valores correctos de temperatura, humedad y presión. Utilice *Python* para calcular.
2. Obtenga la matriz $S^{T}$.
3. Interprete los elementos $s_{24}$ y $s_{31}^{T}$.


# Solución

In [71]:
D = np.array([[20, 28, 22, 26, 24],
              [52, 57, 56, 54, 58],
              [1066, 1068, 1066, 1064, 1068]])

**1)**

In [77]:
S = 0.95 * D

print("Matriz S: \n")
for line in S:
    print("  ".join(str(round(n,2)).rjust(6, " ") for n in line))

Matriz S: 

  19.0    26.6    20.9    24.7    22.8
  49.4   54.15    53.2    51.3    55.1
1012.7  1014.6  1012.7  1010.8  1014.6


**2)**

In [83]:
ST = S.T

print("Matriz S transpuesta: \n")

for line in ST:
    print("  ".join(str(round(n,2)).rjust(6, " ") for n in line))

Matriz S transpuesta: 

  19.0    49.4  1012.7
  26.6   54.15  1014.6
  20.9    53.2  1012.7
  24.7    51.3  1010.8
  22.8    55.1  1014.6


**3)**

In [84]:
print(f"S_24 = {S[1][3]}")
print(f"ST_31 = {ST[2][0]}")

S_24 = 51.3
ST_31 = 20.9


El porcentaje de humedad corregida del día Jueves fue de $51.3\%$.

La temperatura corregida del día Miércoles fue de $20.9 °C$

# Problema 8

Una empresa tecnológica vende software y hardware en tres productos: $P_{1}$, $P_{2}$ y $P_{3}$, en cuatro zonas de la capital: norte, sur, este y oeste. Las ventas del mes, en millones de pesos, están registradas en las matrices $S$ y $H$, para software y hardware respectivamente.

<br>

\begin{split}
S = \begin{pmatrix}
 260 &  480 &  180  &  90  \\
 320 &  440 &  250  &  280 \\
 240 &  360 &  380  &  250  \\
\end{pmatrix} \text{ } H = \begin{pmatrix}
 360 &  250 &  300  &  250  \\
 450 &  500 &  350  &  380 \\
 290 &  300 &  460  &  400  \\
\end{pmatrix}
\end{split}

<br>

1. Calcule en la matriz $T$, del total de ventas de la empresa tecnológica.
2. Un trabajador indica que este mes, la sucursal de la zona sur recaudó más por la venta hardware que software, ¿es correcta su afirmación?
3. Debido al black friday se espera que el siguiente mes, las ventas de software aumenten un $7\%$ y las de hardware aumenten un $3\%$ respecto al mes actual. Encuentra en total de ventas esperadas para el siguiente mes y almacénalo en la matriz $M$. Utiliza *Python* para realizar los cálculos.
4. Interpreta los elementos $t_{24}$ y $m_{31}$ en el contexto del problema.
5. Calcula la matriz $M^{T}$ e interpreta el elemento $m_{41}^{T}$ en el contexto del problema.


# Solución

In [85]:
S = np.array([[260, 480, 180, 90],
              [320, 440, 250, 280],
              [240, 360, 380, 250]])

H = np.array([[360, 250, 300, 250],
              [450, 500, 350, 380],
              [290, 300, 460, 400]])

**1)**

In [86]:
T = S + H
print("Matriz T:\n")

for line in T:
    print("  ".join(str(round(n)) for n in line))

Matriz T:

620  730  480  340
770  940  600  660
530  660  840  650


**2)**

In [89]:
sur_software = 0
for i in range(0,3):
    sur_software += S[i][1]

sur_hardware = 0
for i in range(0,3):
    sur_hardware += H[i][1]

print(f"Las ventas totales de software de la zona sur es {sur_software} millones de pesos")
print(f"Las ventas totales de hardware de la zona sur es {sur_hardware} millones de pesos")

Las ventas totales de software de la zona sur es 1280 millones de pesos
Las ventas totales de hardware de la zona sur es 1050 millones de pesos


El trabajador está equivocado.

**3)**

In [94]:
M = 1.07 * S + 1.03 * H

print("Matriz M: \n")
for line in M:
    print("  ".join(str(round(n,2)) for n in line))

Matriz M: 

649.0  771.1  501.6  353.8
805.9  985.8  628.0  691.0
555.5  694.2  880.4  679.5


**4)**

In [95]:
print(f"T_24 = {T[1][3]}")
print(f"M_31 = {M[2][0]}")

T_24 = 660
M_31 = 555.5


El total de ventas de este mes del producto 2 en la zona oeste es de $660$ millones de pesos.

Se espera que se vendan $555.5$ millones de pesos del producto 3 en la zona norte.

**5)**

In [99]:
MT = M.T

print("Matriz M tranpuesta: \n")
for line in MT:
    print("  ".join(str(round(n,2)) for n in line))

print("\n")
print(f"MT_41 = {MT[3][0]}")

Matriz M tranpuesta: 

649.0  805.9  555.5
771.1  985.8  694.2
501.6  628.0  880.4
353.8  691.0  679.5


MT_41 = 353.8


Se proyecta que se vendan $353.8$ millones de pesos del producto 1 en la zona oeste.